### ISSUE TITLE
[Cache mediator] Unexpired Cached Responses cannot be reincarnated

### ISSUE THREAD

**Issue by VijithaEkanayake** [wso2/product-ei#2222]
**Description:**

The below error is noticed intermittently and ESB responds with an error in this case.

```
TID: [0] [ESB] [2018-05-08 01:30:40,851] ERROR {org.apache.synapse.mediators.base.SequenceMediator} -  Unexpired Cached Responses cannot be reincarnated {org.apache.synapse.mediators.base.SequenceMediator}
java.lang.IllegalStateException: Unexpired Cached Responses cannot be reincarnated
        at org.wso2.caching.CachableResponse.reincarnate(CachableResponse.java:107)
        at org.apache.synapse.mediators.builtin.CacheMediator.processRequestMessage(CacheMediator.java:414)
        at org.apache.synapse.mediators.builtin.CacheMediator.mediate(CacheMediator.java:177)
        at org.apache.synapse.mediators.AbstractListMediator.mediate(AbstractListMediator.java:89)
        at org.apache.synapse.mediators.AbstractListMediator.mediate(AbstractListMediator.java:59)
        at org.apache.synapse.mediators.base.SequenceMediator.mediate(SequenceMediator.java:135)
        at org.apache.synapse.mediators.MediatorWorker.run(MediatorWorker.java:69)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1152)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:622)
        at java.lang.Thread.run(Thread.java:745)
TID: [0] [ESB] [2018-05-08 01:30:40,853]  INFO 
```

**Comment by Nirothipan** [comment]
@GDLMadushanka  We have fixed this in ESB 4.8.1 recently, isn't it?

Does the fix apply to Master branch also?

**Comment by Nirothipan** [comment]
So shall we resolve this ?

**Comment by GDLMadushanka** [comment]
resolving since fixed in ESB 4.8.1
